src/drt/src/gc/FlexGC_main.cpp

```diff
@@ -2299,6 +2299,101 @@ void FlexGCWorker::Impl::checkCutSpacing_spc_diff_layer(
   addMarker(std::move(marker));
 }
 
+void FlexGCWorker::Impl::checkLef58CutSpacing_spc_parallelOverlap(
+    gcRect* rect1,
+    gcRect* rect2,
+    frLef58CutSpacingConstraint* con,
+    const gtl::rectangle_data<frCoord>& markerRect)
+{
+  auto net1 = rect1->getNet();
+  auto net2 = rect2->getNet();
+
+  auto prlX = gtl::delta(markerRect, gtl::HORIZONTAL);
+  auto prlY = gtl::delta(markerRect, gtl::VERTICAL);
+  frCoord prl = std::max(prlX, prlY);
+
+  // skip if no parallel overlap
+  if (prl <= 0) {
+    return;
+    // skip if parallel overlap but shares the same above/below metal
+  }
+
+  box_t queryBox;
+  myBloat(markerRect, 0, queryBox);
+  auto& workerRegionQuery = getWorkerRegionQuery();
+  vector<rq_box_value_t<gcRect*>> result;
+  auto secondLayerNum = rect1->getLayerNum() - 1;
+  if (secondLayerNum >= getTech()->getBottomLayerNum()
+      && secondLayerNum <= getTech()->getTopLayerNum()) {
+    workerRegionQuery.queryMaxRectangle(queryBox, secondLayerNum, result);
+  }
+  secondLayerNum = rect1->getLayerNum() + 1;
+  if (secondLayerNum >= getTech()->getBottomLayerNum()
+      && secondLayerNum <= getTech()->getTopLayerNum()) {
+    workerRegionQuery.queryMaxRectangle(queryBox, secondLayerNum, result);
+  }
+  for (auto& [objBox, objPtr] : result) {
+    // TODO why isn't this auto-converted from Rect to box_t?
+    Rect queryRect(queryBox.min_corner().get<0>(),
+                   queryBox.min_corner().get<1>(),
+                   queryBox.max_corner().get<0>(),
+                   queryBox.max_corner().get<1>());
+    if ((objPtr->getNet() == net1 || objPtr->getNet() == net2)
+        && objBox.contains(queryRect)) {
+      return;
+    }
+  }
+
+  // no violation if spacing satisfied
+  frSquaredDistance reqSpcValSquare
+      = checkLef58CutSpacing_spc_getReqSpcVal(rect1, rect2, con);
+  reqSpcValSquare *= reqSpcValSquare;
+
+  gtl::point_data<frCoord> center1, center2;
+  gtl::center(center1, *rect1);
+  gtl::center(center2, *rect2);
+  frSquaredDistance distSquare = 0;
+  if (con->isCenterToCenter()) {
+    distSquare = gtl::distance_squared(center1, center2);
+  } else {
+    distSquare = gtl::square_euclidean_distance(*rect1, *rect2);
+  }
+  if (distSquare >= reqSpcValSquare) {
+    return;
+  }
+  // no violation if fixed shapes
+  if (rect1->isFixed() && rect2->isFixed()) {
+    return;
+  }
```

Comment:
move this check to the top as there is no reason to do all the other work and then reject it here.